src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/elasticapi/ElasticExtensions.kt

```diff
@@ -210,6 +225,7 @@ class InjectorContextElement(
 
     override fun updateThreadContext(context: CoroutineContext) {
         rolesInjectorHelper.injectRoles(roles)
+        threadContext.putTransient(INDEX_MANAGEMENT_PLUGIN_INTERNAL, internalReq)
```

Comment:
Need to remove this when suspending; also prob need to do a quick sanity check to make sure it isn't already there otherwise you'll get corruption exceptions